python-playwright/Dockerfile

```diff
@@ -1,47 +1,57 @@
-# Use the specified Python version
+# Get the Python version provided as a build argument
 ARG PYTHON_VERSION
-FROM python:${PYTHON_VERSION}-bullseye
 
-LABEL maintainer="support@apify.com" description="Base image for Apify Actors written in Python using Playwright"
+# Extend from the latest Debian and its slim version to keep the image as small as possible
+FROM python:${PYTHON_VERSION}-slim-bullseye
+
+# Add labels to the image to identify it as an Apify Actor
+LABEL maintainer="support@apify.com" \
+      description="Base image for Apify Actors written in Python using Playwright"
+
+# Set the shell to use /bin/bash with specific options (see Hadolint DL4006)
+SHELL ["/bin/bash", "-o", "pipefail", "-c"]
+
+# Get the Apify Python SDK version provided as a build argument
+ARG APIFY_VERSION
+
+# Get the Playwright version provided as a build argument
+ARG PLAYWRIGHT_VERSION
 
 # Don't store bytecode, the Python app will be only run once
-ENV PYTHONDONTWRITEBYTECODE 1
+ENV PYTHONDONTWRITEBYTECODE=1
 
 # Don't buffer output and flush it straight away
-ENV PYTHONUNBUFFERED 1
-
-# Create a virtual environment
-ENV VIRTUAL_ENV=/opt/venv
-RUN python3 -m venv $VIRTUAL_ENV
-ENV PATH="$VIRTUAL_ENV/bin:$PATH"
+ENV PYTHONUNBUFFERED=1
 
 # Disable warnings about outdated pip
 ENV PIP_DISABLE_PIP_VERSION_CHECK=1
 
-# Upgrade pip before installing anything else first
-RUN pip install --upgrade pip
+# Set up XVFB
+# We should use the autodisplay detection as suggested here: https://github.com/microsoft/playwright/issues/2728#issuecomment-678083619
+ENV DISPLAY=:99
+ENV XVFB_WHD=1920x1080x24+32
 
-# Preinstall the latest versions of setuptools and wheel for faster package installs
-RUN pip install --upgrade setuptools wheel
+# This instruction:
+# - Upgrades pip to the latest version
+# - Preinstalls the latest versions of setuptools and wheel to improve package installation speed
+# - Installs the specified version of the Apify Python SDK and Playwright
+RUN pip install --upgrade --no-cache-dir --root-user-action ignore \
```

Comment:
I woult set `--no-cache-dir` and `--root-user-action` through env vars, so that we don't have to do it in every command, and so that users don't have to do it in their Actor Dockerfiles.

```Dockerfile
ENV PIP_NO_CACHE_DIR=1
ENV PIP_ROOT_USER_ACTION=ignore
```

Reply:
That makes sense, thanks